frontend/templates/troubleshooting/server.tsx

```diff
@@ -107,15 +107,11 @@ export const getServerSideProps: GetServerSideProps<TroubleshootingProps> =
       }
 
       const canonicalUrl = new URL(troubleshootingData.canonicalUrl);
-      canonicalUrl.protocol = /localhost/.test(context.req.headers.host || '')
-         ? 'http'
-         : 'https';
-      canonicalUrl.host = context.req.headers.host || canonicalUrl.host;
 
       if (context.resolvedUrl !== canonicalUrl.pathname.toString()) {
          return {
             redirect: {
-               destination: canonicalUrl.toString(),
```

Comment:
Before we perma-redirect, I think we need to compare the *pathname* to the *pathname*. It appears here that we're comparing the *url* (i.e. including query string) to the *pathname*. If we know that `context.resolvedUrl` *doesn't* include the query string, then I'd suggest putting in a local `const currentPathname = context.resolvedUrl`.

Reply:
You're right. I'd suggest turning both in to URL objects and comparing `url.pathname` only (or explicitly checking the params as well)

Reply:
Oh, nice point. I'll switch that.